### PR TITLE
Add Optimum-onnxruntime training blog post

### PR DESCRIPTION
**Pull request by JingyaHuang**
This blog post presents onnxruntime-training integration in Optimum and some benchmark results.

Gently pinging ORT team: @askhade @kspawa

### CODE REVIEW

File: optimum-onnxruntime-training.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: "Optimum+ONNX Runtime - Easier, Faster training for your Hugging Face models"
+thumbnail: blog/assets/optimum_onnxruntime-training/thumbnail.png
+---
+
+# Optimum+ONNX Runtime - Easier, Faster training for your Hugging Face models
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:6]:
```suggestion
# Optimum + ONNX Runtime: Easier, Faster training for your Hugging Face models
```

File: optimum-onnxruntime-training.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            <code>Ashwini</code>
+            <span class="fullname">Ashwini Khade</span>
+            <span class="bg-gray-100 dark:bg-gray-700 rounded px-1 text-gray-600 text-sm font-mono">guest</span>
+        </div>
+    </a>
+</div>
+
+## Introduction
+
+Transformer based models in language, vision and speech are getting larger to support complex multi-modal use cases for the end customer. Increasing model sizes directly impact the resources needed to train these models and scale them as the size increases. Hugging Face and Microsoft’s ONNX Runtime team are working together to build advancements in finetuning large Language, Speech and Vision models. Hugging Face’s Optimum library through its integration with ONNX Runtime for training, provides an open solution to __improve training times by greater than 35%__ for many popular Hugging Face models. We present details of both Hugging Face Optimum and the ONNX Runtime Training ecosystem, with performance numbers highlighting the wins of using the Optimum library.
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:43]:
```suggestion
Transformer based models in language, vision and speech are getting larger to support complex multi-modal use cases for the end customer. Increasing model sizes directly impact the resources needed to train these models and scale them as the size increases. Hugging Face and Microsoft’s ONNX Runtime teams are working together to build advancements in finetuning large Language, Speech and Vision models. Hugging Face’s [Optimum library](https://github.com/huggingface/optimum), through its integration with ONNX Runtime for training, provides an open solution to __improve training times by 35% or more__ for many popular Hugging Face models. We present details of both Hugging Face Optimum and the ONNX Runtime Training ecosystem, with performance numbers highlighting the benefits of using the Optimum library.
```

File: optimum-onnxruntime-training.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Performance results
+
+The chart below shows the phenomenal acceleration we get __from 39% to 130%__ for Hugging Face models with Optimum when __using ONNX Runtime and DeepSpeed ZeRO Stage 1__ for training. The performance measurements were done on selected Hugging Face models with the PyTorch run as the baseline, only ONNX Runtime for training as the second run and ONNX Runtime + DeepSpeed ZeRO Stage 1 as the final run, showing maximum gains. The Optimizer used for the baseline PyTorch runs is the AdamW optimizer and the ORT Training runs use the Fused Adam Optimizer. The runs were performed on a single Nvidia A100 node with 8 GPUs.
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:47]:
```suggestion
The chart below shows impressive acceleration __from 39% to 130%__ for Hugging Face models with Optimum when __using ONNX Runtime and DeepSpeed ZeRO Stage 1__ for training. The performance measurements were done on selected Hugging Face models with PyTorch as the baseline run, only ONNX Runtime for training as the second run, and ONNX Runtime + DeepSpeed ZeRO Stage 1 as the final run, showing maximum gains. The Optimizer used for the baseline PyTorch runs is the AdamW optimizer and the ORT Training runs use the Fused Adam Optimizer. The runs were performed on a single Nvidia A100 node with 8 GPUs.
```

File: optimum-onnxruntime-training.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+Additional details on configuration settings to turn on Optimum for training acceleration can be found [here](https://huggingface.co/docs/optimum/onnxruntime/usage_guides/trainer). The version information used for these runs is as follows:
+```
+PyTorch: 1.14.0.dev20221103+cu116; ORT: 1.14.0.dev20221103001+cu116; DeepSpeed: 0.6.6; HuggingFace: 4.24.0.dev0; Optimum: 1.4.1.dev0; Cuda: 11.6.2
+```
+
+## Optimum Library
+
+Hugging Face is a fast-growing open community and platform aiming to democratize machine learning. As an interactive and community-led company, we extend our coverage from NLP to audio and vision and offers various services in the ML ecosystem to meet the community's needs after the success of the [Transformers library](https://huggingface.co/docs/transformers/index). Now on [Hugging Face Hub](https://huggingface.co/models), there are more than 120K model checkpoints on various machine learning tasks, 18K datasets, and 20K ML demo apps. However, scaling transformer models into production is still a difficult industry challenge. Despite the high accuracy, the training and the inference of transformer-based models could be time-consuming and cost-expensive.
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:60]:
```suggestion
Hugging Face is a fast-growing open community and platform aiming to democratize good machine learning. We extended modalities from NLP to audio and vision, and now cover use cases across Machine Learning to meet our community's needs following the success of the [Transformers library](https://huggingface.co/docs/transformers/index). Now on [Hugging Face Hub](https://huggingface.co/models), there are more than 120K free and accessible model checkpoints for various machine learning tasks, 18K datasets, and 20K ML demo apps. However, scaling transformer models into production is still a challenge for the industry. Despite high accuracy, training and inference of transformer-based models can be time-consuming and expensive.
```

File: optimum-onnxruntime-training.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+To target these needs, Hugging Face built two open-sourced libraries: __Accelerate__ and __Optimum__. While [🤗 Accelerate](https://huggingface.co/docs/accelerate/index) focuses on out-of-box distributed training, [🤗 Optimum](https://huggingface.co/docs/optimum/index), as an extension of transformers, accelerates model training and inference by leveraging the maximum efficiency of users’ targeted hardware. Optimum integrated machine learning accelerators like ONNX Runtime and specialized hardware, so users can benefit from considerable speedup in both training and inference. Besides, Optimum seamlessly integrates other Hugging Face’s tools while inheriting the same ease of use as Transformers. Developers can easily adapt their work to achieve lower latency with less computing power.
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:62]:
```suggestion
To target these needs, Hugging Face built two open-sourced libraries: __Accelerate__ and __Optimum__. While [🤗 Accelerate](https://huggingface.co/docs/accelerate/index) focuses on out-of-the-box distributed training, [🤗 Optimum](https://huggingface.co/docs/optimum/index), as an extension of transformers, accelerates model training and inference by leveraging the maximum efficiency of users’ targeted hardware. Optimum integrated machine learning accelerators like ONNX Runtime and specialized hardware, so users can benefit from considerable speedup in both training and inference. Besides, Optimum seamlessly integrates other Hugging Face’s tools while inheriting the same ease of use as Transformers. Developers can easily adapt their work to achieve lower latency with less computing power.
```

Comment (jeffboudier) [reply optimum-onnxruntime-training.md:62]: I would name and link Habana as an example of "specialized hardware"

File: optimum-onnxruntime-training.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+## ONNX Runtime Training
+
+[ONNX Runtime](https://onnxruntime.ai/) accelerates [large model training](https://onnxruntime.ai/docs/get-started/training-pytorch.html) to speed up throughput by upto 40% standalone and 130% when composed with [DeepSpeed](https://www.deepspeed.ai/tutorials/zero/) for popular HuggingFace transformer based models. ONNX Runtime is already integrated as part of Optimum and enables faster training through Hugging Face’s Optimum training framework.
```
Comment (jeffboudier) [inline optimum-onnxruntime-training.md:66]:
```suggestion
[ONNX Runtime](https://onnxruntime.ai/) accelerates [large model training](https://onnxruntime.ai/docs/get-started/training-pytorch.html) to speed up throughput by up to 40% standalone, and 130% when composed with [DeepSpeed](https://www.deepspeed.ai/tutorials/zero/) for popular HuggingFace transformer based models. ONNX Runtime is already integrated as part of Optimum and enables faster training through Hugging Face’s Optimum training framework.
```

File: optimum-onnxruntime-training.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    </a>
+    <a href="https://github.com/askhade">
+        <img class="avatar avatar-user" src="https://avatars.githubusercontent.com/u/6475296?v=4" title="Gravatar" width="200">
+        <div class="bfc">
+            <code>Ashwini</code>
+            <span class="fullname">Ashwini Khade</span>
+            <span class="bg-gray-100 dark:bg-gray-700 rounded px-1 text-gray-600 text-sm font-mono">guest</span>
+        </div>
+    </a>
+</div>
```
Comment (askhade) [inline optimum-onnxruntime-training.md:39]: Can you add Vincent (https://github.com/iK1D) and Zhijiang (https://github.com/zhijxu-MS) as authors too

Comment (jeffboudier) [reply optimum-onnxruntime-training.md:39]: @askhade you/we can add them in the post header - they need to have a HF account.

Comment (JingyaHuang) [reply optimum-onnxruntime-training.md:39]: Hi @askhade, the easiest way to add the ORT team as authors would be using your Hugging Face Hub accounts. Would it be possible for all authors to create one [here](https://huggingface.co/join) with the avatar and full name you want to show on the author card?

File: optimum-onnxruntime-training.md (hunk 5 above)
Comment (JingyaHuang) [reply optimum-onnxruntime-training.md]: @jeffboudier Added Habana and linked to Habana Gaudi 2 benchmark blog.

File: optimum-onnxruntime-training.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+To target these needs, Hugging Face built two open-sourced libraries: __Accelerate__ and __Optimum__. While [🤗 Accelerate](https://huggingface.co/docs/accelerate/index) focuses on out-of-the-box distributed training, [🤗 Optimum](https://huggingface.co/docs/optimum/index), as an extension of transformers, accelerates model training and inference by leveraging the maximum efficiency of users’ targeted hardware. Optimum integrated machine learning accelerators like ONNX Runtime and specialized hardware like [Intel's Habana GPU](https://huggingface.co/blog/habana-gaudi-2-benchmark), so users can benefit from considerable speedup in both training and inference. Besides, Optimum seamlessly integrates other Hugging Face’s tools while inheriting the same ease of use as Transformers. Developers can easily adapt their work to achieve lower latency with less computing power.
```
Comment (regisss) [inline optimum-onnxruntime-training.md:62]:
```suggestion
To target these needs, Hugging Face built two open-sourced libraries: __Accelerate__ and __Optimum__. While [🤗 Accelerate](https://huggingface.co/docs/accelerate/index) focuses on out-of-the-box distributed training, [🤗 Optimum](https://huggingface.co/docs/optimum/index), as an extension of transformers, accelerates model training and inference by leveraging the maximum efficiency of users’ targeted hardware. Optimum integrated machine learning accelerators like ONNX Runtime and specialized hardware like [Intel's Habana Gaudi](https://huggingface.co/blog/habana-gaudi-2-benchmark), so users can benefit from considerable speedup in both training and inference. Besides, Optimum seamlessly integrates other Hugging Face’s tools while inheriting the same ease of use as Transformers. Developers can easily adapt their work to achieve lower latency with less computing power.
```
It's not a GPU :grin:

File: optimum-onnxruntime-training.md (hunk 7 above)
Comment (askhade) [reply optimum-onnxruntime-training.md:39]: Sure we can create these accounts... Will update this thread once all of us do it.

Comment (askhade) [reply optimum-onnxruntime-training.md:39]: Here is my account: https://huggingface.co/askhade

Comment (askhade) [reply optimum-onnxruntime-training.md:39]: @jeffboudier , @JingyaHuang done. Here are the links for me, Vincent and Zhijiang. Waiting on @kspawa

https://huggingface.co/askhade
https://huggingface.co/weicwang
https://huggingface.co/zhijiang

Comment (JingyaHuang) [reply optimum-onnxruntime-training.md:55]: Thanks @askhade, Zhijiang and Vincent are added as authors!

Comment (askhade) [reply optimum-onnxruntime-training.md:55]: @JingyaHuang Here is Kshama's HF Profile: https://huggingface.co/kshama-msft

Comment (JingyaHuang) [reply optimum-onnxruntime-training.md:55]: @askhade Added! Thanks for creating accounts for the blog post 🙏

**Pull request merged**